lib/buildingsync/makers/workflow_maker.rb

```diff
@@ -431,52 +431,337 @@ def cleanup_larger_files(osw_dir)
+        cleanup_larger_files(osw_dir)
+
+        # find the osw
+        path = File.join(osw_dir, 'out.osw')
+        if !File.exist?(path)
+          puts "Cannot load results for scenario #{scenario_name}, because the osw files does not exist #{path}"
+          next
+        end
+        File.open(path, 'r') do |file|
+          results[scenario_name] = JSON.parse(file.read, symbolize_names: true)
+        end
+        # open results.json to get monthly timeseries
+        # just grabbed openstudio_results
+        path2 = File.join(osw_dir, 'results.json')
+        File.open(path2, 'r') do |file|
+          temp_res = JSON.parse(file.read, symbolize_names: true)
+          monthly_results[scenario_name] = temp_res[:OpenStudioResults]
+        end
+      end
+      return results, monthly_results
+    end
+
+    def delete_scenario_element(results, scenario)
+      package_of_measures = scenario.elements["#{@ns}:ScenarioType"].elements["#{@ns}:PackageOfMeasures"]
+
+      # delete previous results
+      package_of_measures.elements.delete("#{@ns}:AnnualSavingsSiteEnergy")
+      package_of_measures.elements.delete("#{@ns}:AnnualSavingsCost")
+      package_of_measures.elements.delete("#{@ns}:CalculationMethod")
+      package_of_measures.elements.delete("#{@ns}AnnualSavingsByFuels")
+      scenario.elements.delete("#{@ns}AllResourceTotals")
+      scenario.elements.delete("#{@ns}RsourceUses")
+      scenario.elements.delete("#{@ns}AnnualSavingsByFuels")
+
+      result = results[scenario_name]
+      baseline = results['Baseline']
+
+      if result.nil?
+        puts "Cannot load results for scenario #{scenario_name}, because the result is nil"
+        @failed_scenarios << scenario_name
+        return
+      elsif baseline.nil?
+        puts "Cannot load baseline results for scenario #{scenario_name}"
+        @failed_scenarios << scenario_name
+        return
+      end
+
+      if result['completed_status'] == 'Success' || result[:completed_status] == 'Success'
+        # success
+      else
+        @failed_scenarios << scenario_name
+      end
+
+      # preserve existing user defined fields if they exist
+      # KAF: there should no longer be any UDFs
+      user_defined_fields = scenario.elements["#{@ns}:UserDefinedFields"]
+      if user_defined_fields.nil?
+        user_defined_fields = REXML::Element.new("#{@ns}:UserDefinedFields")
+      end
+
+      # delete previous results (if using an old schema)
+      to_remove = []
+      user_defined_fields.elements.each("#{@ns}:UserDefinedField") do |user_defined_field|
+        name_element = user_defined_field.elements["#{@ns}:FieldName"]
+        if name_element.nil?
+          to_remove << user_defined_field
+        elsif /OpenStudio/.match(name_element.text)
+          to_remove << user_defined_field
+        end
+      end
+
+      to_remove.each do |element|
+        user_defined_fields.elements.delete(element)
+      end
+
+      return result, baseline, package_of_measures
```

Comment:
lets split this method in 

- a part that deletes old elements and fields

- a part that gets and returns result, baseline and package_of_measures